File: CartQuestAndroid/app/src/main/java/com/amoghghadge/cartquestandroid/data/repository/UserRepository.kt
package com.amoghghadge.cartquestandroid.data.repository

import com.amoghghadge.cartquestandroid.data.model.User
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import kotlinx.coroutines.tasks.await

class UserRepository {
    private val db = Firebase.firestore

    suspend fun createUserDocument(user: FirebaseUser) {
        val userData = User(
            uid = user.uid,
            email = user.email ?: "",
            displayName = user.displayName ?: "",
            photoUrl = user.photoUrl?.toString() ?: ""
        )
        db.collection("users").document(user.uid).set(userData).await()
    }
}

File: CartQuestiOS/CartQuestiOS/Repositories/UserRepository.swift
import Foundation
import FirebaseAuth
import FirebaseFirestore

// Typealias to disambiguate our app's User model from FirebaseAuth.User
private typealias AppUser = User

class UserRepository {
    private let db = Firestore.firestore()

    func createUserDocument(user: FirebaseAuth.User) async throws {
        let userData = AppUser(
            id: user.uid,
            email: user.email ?? "",
            displayName: user.displayName ?? "",
            photoUrl: user.photoURL?.absoluteString ?? ""
        )
        try await db.collection("users").document(user.uid).setData([
            "uid": userData.id,
            "email": userData.email,
            "displayName": userData.displayName,
            "photoUrl": userData.photoUrl,
            "createdAt": Timestamp(date: userData.createdAt)
        ])
    }
}
